### PR TITLE
changed null check in JSONPathMatcher and JmesPathMatcher to ensure t…

### LINKED ISSUE #1277: Newtonsoft.Json.JsonReaderException: 'Unable to read the JSON string.' when using IStringMatcher with an empty body
### Describe the bug
When using the IStringMatcher (tried it with JmesPathMatcher and JsonPathMatcher) and having an empty body (not null, empty string) an error gets logged (doesnt break execution) to console

### Expected behavior:
No error message is emitted, resulting in clean output.

### Test to reproduce
- Add a mapping with a JmesPathMatcher or JsonPathMatcher
- Request with HTTP Post, without body
- Error gets logged

### PR DESCRIPTION
**Pull request by jollyjoyce1995**
Fixes #1277

### CODE REVIEW

File: src/WireMock.Net/Matchers/JSONPathMatcher.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ public MatchResult IsMatch(string? input)
         var score = MatchScores.Mismatch;
         Exception? exception = null;
 
-        if (input != null)
+        if (!string.IsNullOrWhiteSpace(input))
```
Comment (StefH) [inline src/WireMock.Net/Matchers/JSONPathMatcher.cs:70]: Can you also add a test for empty string based on this test?
https://github.com/WireMock-Net/WireMock.Net/blob/master/test/WireMock.Net.Tests/Matchers/JsonPathMatcherTests.cs#L55

File: src/WireMock.Net/Matchers/JmesPathMatcher.cs [hunk 2]
```diff
@@ -74,7 +74,7 @@ public MatchResult IsMatch(string? input)
         var score = MatchScores.Mismatch;
         Exception? exception = null;
 
-        if (input != null)
+        if (!string.IsNullOrWhiteSpace(input))
```
Comment (StefH) [inline src/WireMock.Net/Matchers/JmesPathMatcher.cs:77]: Same for this.

See
https://github.com/WireMock-Net/WireMock.Net/blob/master/test/WireMock.Net.Tests/Matchers/JmesPathMatcherTests.cs#L55

**Review by StefH — CHANGES_REQUESTED** [review]

**Comment by StefH** [comment]
@jollyjoyce1995
Can you please check my comments?
